mitmproxy/addons/cut.py

```diff
@@ -28,6 +28,14 @@ def is_addr(v):
 
 
 def extract(cut: str, f: flow.Flow) -> str | bytes:
+    if (
+        hasattr(f, "websocket")
+        and f.websocket is not None
+        and (cut == "response.content" or cut == "request.content")
+    ):
```

Comment:
```suggestion
    # Hack for https://github.com/mitmproxy/mitmproxy/issues/6721:
    # Make "save body" keybind work for WebSocket flows.
    # Ideally the keybind would be smarter and this here can get removed.
    if (
        getattr(f, "websocket", None)
        and cut == "response.content"
    ):
```

Reply:
```
hasattr(f, "websocket")
and f.websocket is not None
```
Both above lines seems to be required, there is a test which fails otherwise. In the test scenario, websocket attribute exists but is `None`. Your change includes getattr(f, "websocket", None), but that just seems checks if the websocket attribute is present (otherwise returns None as default)

Reply:
> otherwise returns None as default

That is true, but `bool(None)` evaluates to false and the condition fails. :)


The intention behind just doing it for response.content was to limit the impact of this workaround, but I don't feel particularly strongly about this one. Let's keep it as-is if you think that's better. :)